README.md

```diff
@@ -240,22 +240,527 @@ For complete documentation of the `mp-build` tool see the [MP Build Tool Documen
+After also removing the 'ID' parameter from the sample adapter, the method should look
+similar to this:
+
+```python
+def get_adapter_definition() -> AdapterDefinition:
+    logger.info("Starting 'Get Adapter Definition'")
+    definition = AdapterDefinition(ADAPTER_KIND, ADAPTER_NAME)
+
+    definition.define_string_parameter("host", "MySQL Host")
+    definition.define_int_parameter("port", "Port", default=3306)
+
+    credential = definition.define_credential_type("mysql_user", "MySQL User")
+    credential.define_string_parameter("username", "Username")
+    credential.define_password_parameter("password", "Password")
+
+    # The key 'container_memory_limit' is a special key that is read by the VMware Aria 
+    # Operations collector to determine how much memory to allocate to the docker 
+    # container running this adapter. It does not need to be read inside the adapter 
+    # code.
+    definition.define_int_parameter(
+        "container_memory_limit",
+        label="Adapter Memory Limit (MB)",
+        description="Sets the maximum amount of memory VMware Aria Operations can "
+                    "allocate to the container running this adapter instance.",
+        required=True,
+        advanced=True,
+        default=1024,
+    )
+
+    # This Adapter has no object types directly, rather it co-opts object types that
+    # are part of the MySQL MP to add additional metrics. As such, we can't define
+    # any object types here, because they're already defined in the MySQL MP with a
+    # different adapter type.
+
+    logger.info("Finished 'Get Adapter Definition'")
+    logger.debug(f"Returning adapter definition: {definition.to_json()}")
+    return definition
+```
+
+The adapter definition is also where objects and metrics are defined, however we are 
+only allowed to define objects and metrics that are a part of our adapter type. Because
+extensions modify objects that are part of a different adapter type, we can't add them.
+This means that we can't set metric metadata like 'units', 'labels', etc that we would
+generally be able to set.
+
+#### Modify the `test` method to create a MySQL connection and run a query
+
+Now that we've defined our parameters, we can try to connect and run a test query.
+We will do this in the `test` method. Notice this takes an `AdapterInstance` as a 
+parameter. All the parameters from the definition will be present in this Adapter 
+Instance. We can access them like this:
+
+```python
+    hostname = adapter_instance.get_identifier_value("host")
+    port = int(adapter_instance.get_identifier_value("port", "3306"))
+    username = adapter_instance.get_credential_value("username")
+    password = adapter_instance.get_credential_value("password")
+```
+
+We can then use them to connect to MySQL and run a test query:
+
+```python
+    connection = mysql.connector.connect(
+        host=hostname,
+        port=port,
+        user=username,
+        password=password,
```

Comment:
extra comma

Reply:
That's actually valid and the preferred style when arguments are each on their own line.